UnityProject/Assets/Scripts/UI/PopUps/GUI_Rename.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class GUI_Rename : NetTab
+{
+
+	public InputField textField;
+	public ContentSizeFitter contentSizeFitter;
+
+	private const int MAX_NAME_LENGTH = 42;
+
+	public override void OnEnable()
+	{
+		base.OnEnable();
+		StartCoroutine(WaitForProvider());
+		textField.interactable = false;
+	}
+
+	IEnumerator WaitForProvider()
+	{
+		while (Provider == null)
+		{
+			yield return WaitFor.EndOfFrame;
+		}
+		RefreshText();
+	}
+
+	public override void RefreshTab()
+	{
+		RefreshText();
+		base.RefreshTab();
+	}
+
+	public void RefreshText()
+	{
+		if (Provider != null)
+		{
+			textField.text = Provider.GetComponent<Renameable>().GetCustomName();
+		}
+	}
+
+	public void CloseDialog()
+	{
+		ControlTabs.CloseTab(Type, Provider);
+	}
+
+	public void OnEditStart()
+	{
+		textField.interactable = true;
+		textField.ActivateInputField();
+
+		UIManager.IsInputFocus = true;
+		CheckForInput();
+	}
+
+	//Safety measure:
+	private async void CheckForInput()
+	{
+		await Task.Delay(500);
+		if (!textField.isFocused)
+		{
+			UIManager.IsInputFocus = false;
+		}
+	}
+
+	//Request an edit from server:
+	public void OnTextEditEnd()
+	{
+		var customName = textField.text;
+		if (customName.Length > MAX_NAME_LENGTH)
+		{
+			customName = customName.Substring(0, MAX_NAME_LENGTH);
+		}
+
+		PlayerManager.LocalPlayerScript.playerNetworkActions.CmdRequestRename(Provider.gameObject, customName);
```

Comment:
~~It's a bit weird: using NetTabs, but running clientside code here only while they're designed for server authoritative NetElements.
Well, it works as is, but generally it's expected to use NetSubmitButton for sending text from your InputField to some serverside method (you can check TabComms – AnnouncementPage)~~
...
I just realized that you copied Paper code, never really looked at it before. It's acceptable to leave it as is, papers are a bit of special case. NetTabs are generally more machine-oriented.

Reply:
I did just copy paper code :) Was not aware that NetTabs were more for machines. For future UI popups (file folders and such), is there a different system to use?

Reply:
I'm afraid there aren't any other systems for that aside pure Unity UI. Inventory stripping should work really nicely with NetTabs though